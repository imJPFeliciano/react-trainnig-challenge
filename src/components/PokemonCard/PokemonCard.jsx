import PropTypes from 'prop-types';
import styled from 'styled-components';

const PokemonCardWrapper = styled.div`
  display: flex;
  border: 1px solid black;
  width: 50%;

  img {
    width: 200px;
  }
`;

const PokemonCard = ({ name, sprites }) => (
  <PokemonCardWrapper data-testid="pokemon-card">
    <span>
      Pokemon Name:
      <b data-testid="pokemon-name">{name}</b>
    </span>
    <img data-testid="pokemon-image" src={sprites.front_default} alt="pokemon" />
  </PokemonCardWrapper>
);

PokemonCard.propTypes = {
  name: PropTypes.string.isRequired,
  sprites: PropTypes.shape({
    front_default: PropTypes.string.isRequired,
    back_default: PropTypes.string.isRequired,
  }),
};

PokemonCard.defaultProps = {
  sprites: {
    front_default: '',
    back_default: '',
  },
};

export default PokemonCard;

