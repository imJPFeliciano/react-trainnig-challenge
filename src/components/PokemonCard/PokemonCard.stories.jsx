import React from 'react';
import PokemonCard from './PokemonCard';

export default {
    title: 'PokemonCard',
    component: PokemonCard,
}

const Template = (args) => <PokemonCard {...args} />;

export const DefaultCard = Template.bind({});
DefaultCard.args = {
    name: 'Pikachu',
    sprites: {
        front_default: 'https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/b3997ebc-1e83-48f2-9854-eb28c8c6a97b/debkf0q-ea4fb2bf-81e6-447f-99ff-3ab9bd7aad36.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcL2IzOTk3ZWJjLTFlODMtNDhmMi05ODU0LWViMjhjOGM2YTk3YlwvZGVia2YwcS1lYTRmYjJiZi04MWU2LTQ0N2YtOTlmZi0zYWI5YmQ3YWFkMzYucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.PHxKCx2V24Lng7AyDk6SuDsnuby-nHpRD0E5O4IY-jY',
        back_default: 'https://www.kindpng.com/picc/m/79-794624_pikachu-back-png-transparent-png.png'
    }
}