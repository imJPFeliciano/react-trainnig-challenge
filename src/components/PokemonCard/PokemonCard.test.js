import { render, screen } from '@testing-library/react';
import PokemonCard from './PokemonCard';

const data = {
  name: "pikachu",
  sprites: {
      front_default: "test",
      back_default: "test"
  }
};

test('renders pokemon card properly', () => {
  render(<PokemonCard {...data} />);
  const pokemonTitleElement = screen.getByText(/Pokemon/i);
  expect(pokemonTitleElement).toBeInTheDocument();
});

test('renders pokemon name properly', () => {
  render(<PokemonCard {...data} />);
  const pokemonName = screen.getByTestId('pokemon-name');
  expect(pokemonName).toBeInTheDocument();
  expect(pokemonName).toHaveTextContent(data.name);
});

test('renders pokemon image properly', () => {
  render(<PokemonCard {...data} />);
  const pokemonImage = screen.getByTestId('pokemon-image');
  expect(pokemonImage).toBeInTheDocument();
  expect(pokemonImage.src).toContain(data.sprites.front_default);
});

test('matches the snapshot', () => {
  render(<PokemonCard {...data} />);
  const pokemonCard = screen.getByTestId('pokemon-card');
  expect(pokemonCard).toMatchSnapshot();
});

