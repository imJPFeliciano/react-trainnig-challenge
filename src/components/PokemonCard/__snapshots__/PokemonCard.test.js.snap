// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  class="sc-bdvvtL gDGRzA"
  data-testid="pokemon-card"
>
  <span>
    Pokemon Name:
    <b
      data-testid="pokemon-name"
    >
      pikachu
    </b>
  </span>
  <img
    alt="pokemon"
    data-testid="pokemon-image"
    src="test"
  />
</div>
`;
